app/controllers/alchemy/pages_controller_extension.rb

```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+
+Alchemy::PagesController.module_exec do
+  ##
+  # Extends the render_page method in order to allow json responses.
+  #
+  def render_page
```

Comment:
Instead of duplicating the original content of `#render_page`, alias that method as something else, then override it to provide the JSON response, but delegate to the aliased original for other formats.

Reply:
I've updated this, not sure if exactly how you had in mind.